Add Workarea 3.5 release notes

chriscressman opened the pull request:
(This is an initial commit to provide a starting point for the release
notes and will be edited and expanded in future commits)

* Add 3.5 release notes doc
* Link to 3.5 release notes doc from release notes index
* Rename and modify 3.5 upgrade guide for consistency with 3.4 upgrade guide
* Cross-reference 3.5 release notes and upgrade guide

chriscressman commented:
For this PR I'm primarily looking for feedback on the list of changes, like, "you forgot x", or "make sure you say something about y" (no need to mention anything covered in the upgrade guide; I'll bring that stuff in soon)

bencrouse commented on docs/source/release-notes/workarea-3-5-0.html.md:
```diff
@@ -0,0 +1,268 @@
+* Removes release undo functionality in favor of using separate "undo" releases, and adds a workflow for creating them
+* Restores the releases calendar feature in the Admin
+* Allows data file importing in releases
+* Allows featured product changes in releases
```
(minor) This may be nitpick, but featured product changes in releases were always available, the changes to the search index allow accurate previewing of this now.

bencrouse commented on docs/source/release-notes/workarea-3-5-0.html.md:
```diff
@@ -0,0 +1,268 @@
+----------------------------------------------------------------------
+
+* Adds segmentation feature to the base platform
+* Adds segment-based metrics and insights
```
We'll want another point here to indicate that you can set things things active per-segment to allow to segmented site experiences.

chriscressman commented:
* Re-organized changes into major and minor sections
* Collapsed pull request links to take up a lot less vertical space
* Addressed feedback
* Added latest changes from master
* Improved some descriptions

Still TODO:

* Finish descriptions
* Add overall summary
* Update/maintain until 3.5 code freeze

tubbo commented on docs/source/upgrade-guides/workarea-3-5-0.html.md:
```diff
@@ -1,11 +1,14 @@
-title: Upgrading from Workarea v3.4 to v3.5
-excerpt: A catalog of changes to watch out for while upgrading between minor versions
+title: Workarea 3.5.0 Upgrade Guide
+excerpt: Upgrade guide for Workarea 3.5.0
```
Is this a placeholder or will this `excerpt` give a bit more info than the title?

chriscressman commented:
This is ready for final review and merge. I'll update these docs again after 3.5 is in code freeze.
